Cabal/tests/UnitTests/Distribution/Version.hs

```diff
@@ -104,23 +104,22 @@ versionTests =
 --     -- , property prop_parse_disp5
 --   ]
 
-#if !MIN_VERSION_QuickCheck(2,9,0)
+adjustSize :: (Int -> Int) -> Gen a -> Gen a
+adjustSize adjust gen = sized (\n -> resize (adjust n) gen)
+
 instance Arbitrary Version where
   arbitrary = do
     branch <- smallListOf1 $
                 frequency [(3, return 0)
                           ,(3, return 1)
                           ,(2, return 2)
                           ,(1, return 3)]
-    return (Version branch []) -- deliberate []
+    return (mkVersion branch) -- deliberate []
```

Comment:
Drop the `-- deliberate []` comment. That no longer applies with the new representation.